app/assets/stylesheets/common/base/menu-panel.scss

```diff
@@ -689,7 +691,7 @@ body.footer-nav-ipad {
   background-color: black;
   --opacity: 0.5;
   opacity: var(--opacity);
-  top: 0;
+  top: var(--header-top);
```

Comment:
I could be wrong here, but doesn't this match `--footer-nav-height` + safe-area-offset? I.e. do we need the extra JS calculation in the site header component? 

Another option would be to have the cloak element be 100% height/width but adjust its z-index so it is below the sidebar and nav. (And even further, sorry, thinking out loud, this can also be a pseudo-element.)

Reply:
I don't quite follow. Which JS calculation are you hoping to remove from the site-header component?

`--header-top` does not include the footer-nav-height. In fact, in a default Discourse install, `--header-top` will always be 0. It only has a non-zero value when custom themes introduce stuff inside `.header-wrap` before the `<header>` element. (this was existing logic that I've ported from the JS implementation as-is)

Reply:
I was referring to the calculation for `--header-top`, I guess I got confused by the name of the property. You are correct, this is not new, but do we need to calculate it here? Even if a theme adds elements above `.header-wrap`, we'd still want `header-cloak` to overlay the custom elements when one of the menus is expanded.

Reply:
> we'd still want header-cloak to overlay the custom elements when one of the menus is expanded.

That would've been my guess as well, but that's not what happens in `main` at the moment. I've tried to avoid changing any behaviour in this PR.

There is a commit [here](https://github.com/discourse/discourse/commit/49fbedc445da74fae26b93ed97d1b45adfcb9f0e) and some discussion on Meta [here](https://meta.discourse.org/t/problems-with-full-height-swipe-able-menus/105089?u=david). Looks like the slide-in menus and the header-cloak are intended to be shown below any custom header content.

Reply:
Yeah, fair enough, no need to hold up the PR. That meta discussion from 2018 is a little odd... the end screenshot is not what I would expect the app to do...